db/change_listener.go

```diff
@@ -308,7 +309,7 @@ func (listener *changeListener) NewWaiter(keys []string, trackUnusedSequences bo
 		listener:                  listener,
 		keys:                      keys,
 		lastCounter:               listener.CurrentCount(keys),
-		lastTerminateCheckCounter: listener.terminateCheckCounter,
+		lastTerminateCheckCounter: listener.terminateCheckCounter.Load(),
```

Comment:
I think your PR comment suggests that that using listener.L.Lock to safely fetch the value of terminateCheckCounter here will deadlock.  Can you clarify why that's the case, or what the issue of that approach is?  It would be preferable to avoid acquiring two locks in changeListener.Wait, since that's used highly concurrently.

Reply:
I misanalysed the situation but it was caught in tests. I pushed an update that removes atomic and just acquires the lock once for `NewWaiter` instead of once for `listener._currentCount` and `listener.terminateCheckCounter`.

As far as I could tell, everywhere else we already acquire the lock on read or write of `listener._terminateCheckCounter`.